install/zowe-install.sh

```diff
@@ -10,199 +10,277 @@
+    ls -l $INSTALL_DIR/fingerprint/*  >> $LOG_FILE
+    mkdir -p  $ZOWE_ROOT_DIR/fingerprint
+    chmod a+x $ZOWE_ROOT_DIR/fingerprint
+    echo "Copying `ls $INSTALL_DIR/fingerprint/*` into "$ZOWE_ROOT_DIR/fingerprint >> $LOG_FILE
+    cp $INSTALL_DIR/fingerprint/* $ZOWE_ROOT_DIR/fingerprint
+    chmod a+r $ZOWE_ROOT_DIR/fingerprint/*
+  else
+    echo "OK: No fingerprint"
+    echo "OK: No fingerprint in install directory $INSTALL_DIR, create it with zowe-generate-checksum.sh" >> $LOG_FILE
+  fi
+}
 
+copy_runtime_support_files() {
+  echo "Copying the opercmd into "$ZOWE_ROOT_DIR/scripts/internal >> $LOG_FILE
+  cp $INSTALL_DIR/scripts/opercmd $ZOWE_ROOT_DIR/scripts/internal/opercmd
+  cp $INSTALL_DIR/scripts/ocopyshr.sh $ZOWE_ROOT_DIR/scripts/internal/ocopyshr.sh
+  cp $INSTALL_DIR/scripts/ocopyshr.clist $ZOWE_ROOT_DIR/scripts/internal/ocopyshr.clist
+  echo "Copying the run-zowe.sh into "$ZOWE_ROOT_DIR/scripts/internal >> $LOG_FILE
+  chmod -R 755 $ZOWE_ROOT_DIR/scripts/internal
+
+  mkdir -p ${ZOWE_ROOT_DIR}/bin
+  cp -r $INSTALL_DIR/bin/. $ZOWE_ROOT_DIR/bin
+  chmod -R 755 $ZOWE_ROOT_DIR/bin
+
+  # Prepare utils directory 
+  mkdir -p ${ZOWE_ROOT_DIR}/scripts/utils
+  cp $INSTALL_DIR/scripts/instance.template.env ${ZOWE_ROOT_DIR}/scripts/instance.template.env
+  cp -r $INSTALL_DIR/scripts/utils/. ${ZOWE_ROOT_DIR}/scripts/utils
+  cp $INSTALL_DIR/scripts/tag-files.sh $ZOWE_ROOT_DIR/scripts/utils/tag-files.sh
+}
 
-if [ `uname` = "OS/390" ]; then
-if [[ -z "$DSN_PREFIX" ]]
-then
-  echo "-h parameter not set. Usage: $0 -i zowe_install_path -h zowe_dsn_prefix"
-  exit 1
-else
-  ZOWE_DSN_PREFIX=$DSN_PREFIX
-fi
-fi
+copy_workflow() {
+  mkdir -p ${ZOWE_ROOT_DIR}/workflows
+  cp -r $INSTALL_DIR/files/workflows/. $ZOWE_ROOT_DIR/workflows/
+}
 
-echo "Beginning install of Zowe ${ZOWE_VERSION} into directory " $ZOWE_ROOT_DIR
+install_mvs() {
+  if [ "${RUN_ON_ZOS}" = "true" ]; then
+    echo "Creating MVS artefacts SZWEAUTH and SZWESAMP" >> $LOG_FILE
+    . $INSTALL_DIR/scripts/zowe-install-MVS.sh
+  fi
+}
 
-NEW_INSTALL="true"
-
-# warn about any prior installation
-count_children_in_directory ${ZOWE_ROOT_DIR}
-root_dir_existing_children=$?
-if [[ ${root_dir_existing_children} -gt 0 ]]; then
-    if [[ -f "${ZOWE_ROOT_DIR}/manifest.json" ]]
-    then
-        OLD_VERSION=$(cat ${ZOWE_ROOT_DIR}/manifest.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[",]//g' | tr -d '[[:space:]]')
-        NEW_INSTALL="false"
-        echo "  $ZOWE_ROOT_DIR contains version ${OLD_VERSION}. Updating this install to version ${ZOWE_VERSION}."
-        echo "  Backing up previous Zowe runtime files to ${ZOWE_ROOT_DIR}.${OLD_VERSION}.bak."
-        mv ${ZOWE_ROOT_DIR} ${ZOWE_ROOT_DIR}.${OLD_VERSION}.bak
+install_buildin_components() {
+  component_list="jobs-api files-api api-catalog discovery gateway caching-service apiml-common-lib explorer-ui-server explorer-jes explorer-mvs explorer-uss"
```

Comment:
no app-server or zss here either?

Reply:
Yes for now. We will need to convert app-server/zss packages to follow manifest definition before the installer can proper handle it. So for now, the zlux installation is still in old way.